bindings/rust/s2n-tls/src/testing.rs

```diff
@@ -0,0 +1,202 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+use bytes::Bytes;
+use core::task::Poll;
+use std::collections::VecDeque;
+use crate::testing::s2n_tls::Harness;
+use crate::raw::securitypolicy::*;
+use crate::raw::config::*;
+
+pub mod s2n_tls;
+
+type Error = Box<dyn std::error::Error>;
+type Result<T, E = Error> = core::result::Result<T, E>;
+
+const SAMPLES: usize = 100;
```

Comment:
```suggestion
/// The number of iterations that will be executed until the handshake exits with an error
///
/// This is to prevent endless looping without making progress on the connection.
const SAMPLES: usize = 100;
```